test/functional/rpc_users.py

```diff
@@ -76,32 +78,37 @@ def test_auth(self, node, user, password):
         assert_equal(200, call_with_auth(node, user, password).status)
 
         self.log.info('Wrong...')
-        assert_equal(401, call_with_auth(node, user, password+'wrong').status)
+        assert_equal(401, call_with_auth(node, user, password + 'wrong').status)
 
         self.log.info('Wrong...')
-        assert_equal(401, call_with_auth(node, user+'wrong', password).status)
+        assert_equal(401, call_with_auth(node, user + 'wrong', password).status)
 
         self.log.info('Wrong...')
-        assert_equal(401, call_with_auth(node, user+'wrong', password+'wrong').status)
+        assert_equal(401, call_with_auth(node, user + 'wrong', password + 'wrong').status)
 
     def run_test(self):
-
-        ##################################################
-        # Check correctness of the rpcauth config option #
-        ##################################################
+        self.log.info('Check correctness of the rpcauth config option')
         url = urllib.parse.urlparse(self.nodes[0].url)
 
         self.test_auth(self.nodes[0], url.username, url.password)
         self.test_auth(self.nodes[0], 'rt', self.rtpassword)
         self.test_auth(self.nodes[0], 'rt2', self.rt2password)
         self.test_auth(self.nodes[0], self.user, self.password)
 
-        ###############################################################
-        # Check correctness of the rpcuser/rpcpassword config options #
-        ###############################################################
+        self.log.info('Check correctness of the rpcuser/rpcpassword config options')
         url = urllib.parse.urlparse(self.nodes[1].url)
 
         self.test_auth(self.nodes[1], self.rpcuser, self.rpcpassword)
 
+        self.log.info('Check that failure to write cookie file will abort the node gracefully')
+        self.stop_node(0)
+        cookie_file = os.path.join(get_datadir_path(self.options.tmpdir, 0), self.chain, '.cookie.tmp')
+        with open(cookie_file, 'w', encoding='utf8') as f:
+            f.write('readonly')
+        os.chmod(cookie_file, S_IREAD | S_IRGRP | S_IROTH)
```

Comment:
In commit "http: Avoid crash when g_thread_http was never started" (faf838bf1cae719c46211ea603848f367ee427cf)

Not sure, but it might be more portable to make cookie_file a non-file path e.g. os.mkdir(cookie_file), because I'm not sure different file systems like fat will have good enough support for chmod.

Possible travis errors like https://travis-ci.org/github/bitcoin/bitcoin/jobs/688369052#L5050 might be happening for this reason

Reply:
Thanks for finding out why this failed in travis, but not locally